# Déjame escuchar la música

# Contenido <a id='back'></a>

* [Introducción](#intro)
* [Etapa 1. Descripción de los datos](#data_review)
    * [Conclusiones](#data_review_conclusions)
* [Etapa 2. Preprocesamiento de datos](#data_preprocessing)
    * [2.1 Estilo del encabezado](#header_style)
    * [2.2 Valores ausentes](#missing_values)
    * [2.3 Duplicados](#duplicates)
    * [2.4 Conclusiones](#data_preprocessing_conclusions)
* [Etapa 3. Prueba de hipótesis](#hypothesis)
    * [3.1 Hipótesis 1: actividad de los usuarios y las usuarias en las dos ciudades](#activity)
* [Conclusiones](#end)


## Introducción <a id='intro'></a>
Como analista de datos, tu trabajo consiste en analizar datos para extraer información valiosa y tomar decisiones basadas en ellos. Esto implica diferentes etapas, como la descripción general de los datos, el preprocesamiento y la prueba de hipótesis.

Siempre que investigamos, necesitamos formular hipótesis que después podamos probar. A veces aceptamos estas hipótesis; otras veces, las rechazamos. Para tomar las decisiones correctas, una empresa debe ser capaz de entender si está haciendo las suposiciones correctas.

En este proyecto, compararás las preferencias musicales de las ciudades de Springfield y Shelbyville. Estudiarás datos reales de transmisión de música online para probar la hipótesis a continuación y comparar el comportamiento de los usuarios y las usuarias de estas dos ciudades.

### Objetivo:
Prueba la hipótesis:
1. La actividad de los usuarios y las usuarias difiere según el día de la semana y dependiendo de la ciudad.


### Etapas
Los datos del comportamiento del usuario se almacenan en el archivo `/datasets/music_project_en.csv`. No hay ninguna información sobre la calidad de los datos, así que necesitarás examinarlos antes de probar la hipótesis.

Primero, evaluarás la calidad de los datos y verás si los problemas son significativos. Entonces, durante el preprocesamiento de datos, tomarás en cuenta los problemas más críticos.

Tu proyecto consistirá en tres etapas:
 1. Descripción de los datos.
 2. Preprocesamiento de datos.
 3. Prueba de hipótesis.








[Volver a Contenidos](#back)

## Etapa 1. Descripción de los datos <a id='data_review'></a>

Abre los datos y examínalos.

Necesitarás `pandas`, así que impórtalo.

In [1]:
# Importar pandas
import pandas as pd 

Lee el archivo `music_project_en.csv` de la carpeta `/datasets/` y guárdalo en la variable `df`:

In [2]:
# Leer el archivo y almacenarlo en df
df = pd.read_csv('/datasets/music_project_en.csv')

Muestra las 10 primeras filas de la tabla:

In [3]:
# Obtener las 10 primeras filas de la tabla df
df.head(10)

userID                        Track            artist   genre  \
0  FFB692EC            Kamigata To Boots  The Mass Missile    rock   
1  55204538  Delayed Because of Accident  Andreas Rönnberg    rock   
2    20EC38            Funiculì funiculà       Mario Lanza     pop   
3  A3DD03C9        Dragons in the Sunset        Fire + Ice    folk   
4  E2DC1FAE                  Soul People        Space Echo   dance   
5  842029A1                       Chains          Obladaet  rusrap   
6  4CB90AA5                         True      Roman Messer   dance   
7  F03E1C1F             Feeling This Way   Polina Griffith   dance   
8  8FA1D3BE                     L’estate       Julia Dalia  ruspop   
9  E772D5C0                    Pessimist               NaN   dance   

        City        time        Day  
0  Shelbyville  20:28:33  Wednesday  
1  Springfield  14:07:09     Friday  
2  Shelbyville  20:58:07  Wednesday  
3  Shelbyville  08:37:09     Monday  
4  Springfield  08:34:34     Monday  
5  Shelbyville  13:09:41     Friday  
6  Springfield  13:00:07  Wednesday  
7  Springfield  20:47:49  Wednesday  
8  Springfield  09:17:40     Friday  
9  Shelbyville  21:20:49  Wednesday

Obtén la información general sobre la tabla con un comando. Conoces el método que muestra la información general que necesitamos.

In [4]:
# Obtener la información general sobre nuestros datos
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65079 entries, 0 to 65078
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0     userID  65079 non-null  object
 1   Track     63736 non-null  object
 2   artist    57512 non-null  object
 3   genre     63881 non-null  object
 4     City    65079 non-null  object
 5   time      65079 non-null  object
 6   Day       65079 non-null  object
dtypes: object(7)
memory usage: 3.5+ MB
None


Estas son nuestras observaciones sobre la tabla. Contiene siete columnas. Almacenan los mismos tipos de datos: `object`.

Según la documentación:
- `' userID'`: identificador del usuario o la usuaria;
- `'Track'`: título de la canción;
- `'artist'`: nombre del artista;
- `'genre'`: género de la pista;
- `'City'`: ciudad del usuario o la usuaria;
- `'time'`: la hora exacta en la que se reprodujo la canción;
- `'Day'`: día de la semana.

Podemos ver tres problemas con el estilo en los encabezados de la tabla:
1. Algunos encabezados están en mayúsculas, otros en minúsculas.
2. Hay espacios en algunos encabezados.
3. No hay separación de las palabras con '_' para los encabezados que tienen dos palabras.




### Escribe observaciones de tu parte. Estas son algunas de las preguntas que pueden ser útiles: <a id='data_review_conclusions'></a>

`1.   ¿Qué tipo de datos tenemos a nuestra disposición en las filas? ¿Y cómo podemos entender lo que almacenan las columnas?`

Las filas tienen información de los títulos de canciones,con su artista, el género de la música , la ciudad de cada usuario, el día y la hora que la música se reprodujo. 
Los que alamacean las columnas son el userID: es el identificador clave de cada usurario, Track: título de cada canción , artist: nombre del artista, genre: género de la música, City: ciudad de cada usuario, time: hora exacta en que se reprodujo la canción, Day: día de la semana en que se reproduce la canción. Según la información obtenida con el método info() todos lo datos de este Data Frame son de tipo object que representan cadenas de texto.


`2.   ¿Hay suficientes datos para proporcionar respuestas a nuestra hipótesis o necesitamos más información?`

Al pasar por un rápido escaner de revisión sí hay datos suficientes para probar la hipótesis de las ciudades que son objeto de estudio. Es importante realizar un análisis más a profundidad conforme avance con el proyecto para confirmar lo que se menciona en primera instancia. 

`3.   ¿Notaste algún problema en los datos, como valores ausentes, duplicados o tipos de datos incorrectos?`
Existen los primeros problemas que se pudo dectectar:
1. Valores ausentes: Las columnas 'Track', 'artist' y 'genre' tienen valores ausentes, lo que podría afectar el análisis correcto de las preferencias musicales por género o artista.
2. Duplicados: Podrían existir duplicados explícitos o implícitos.
3. Nombres de las columnas no tienen un formato establecido y consistente.

[Volver a Contenidos](#back)

## Etapa 2. Preprocesamiento de datos <a id='data_preprocessing'></a>

El objetivo aquí es preparar los datos para que sean analizados.
El primer paso es resolver cualquier problema con los encabezados. Luego podemos avanzar a los valores ausentes y duplicados. Empecemos.

Corrige el formato en los encabezados de la tabla.


### Estilo del encabezado <a id='header_style'></a>
Muestra los encabezados de la tabla (los nombres de las columnas):

In [5]:
# Muestra los nombres de las columnas
print(df.columns)

Index(['  userID', 'Track', 'artist', 'genre', '  City  ', 'time', 'Day'], dtype='object')


Cambia los encabezados de la tabla de acuerdo con las reglas del buen estilo:
* Todos los caracteres deben ser minúsculas.
* Elimina los espacios.
* Si el nombre tiene varias palabras, utiliza snake_case.

Anteriormente, aprendiste acerca de la forma automática de cambiar el nombre de las columnas. Vamos a aplicarla ahora. Utiliza el bucle for para iterar sobre los nombres de las columnas y poner todos los caracteres en minúsculas. Cuando hayas terminado, vuelve a mostrar los encabezados de la tabla:

In [6]:
# Bucle en los encabezados poniendo todo en minúsculas

new_columns = []
for column in df.columns:
    new_columns.append(column.lower())  

df.columns = new_columns

print(df.columns)

Index(['  userid', 'track', 'artist', 'genre', '  city  ', 'time', 'day'], dtype='object')


Ahora, utilizando el mismo método, elimina los espacios al principio y al final de los nombres de las columnas e imprime los nombres de las columnas nuevamente:

In [7]:
# Bucle en los encabezados eliminando los espacios
new_columns = []
for column in df.columns:
    new_columns.append(column.strip ())  

df.columns = new_columns

print(df.columns)

Index(['userid', 'track', 'artist', 'genre', 'city', 'time', 'day'], dtype='object')


Necesitamos aplicar la regla de snake_case a la columna `userid`. Debe ser `user_id`. Cambia el nombre de esta columna y muestra los nombres de todas las columnas cuando hayas terminado.

In [8]:
# Cambiar el nombre de la columna "userid"
df.rename(columns={'userid': 'user_id'}, inplace=True)
print(df.columns)

Index(['user_id', 'track', 'artist', 'genre', 'city', 'time', 'day'], dtype='object')


Comprueba el resultado. Muestra los encabezados una vez más:

In [9]:
# Comprobar el resultado: la lista de encabezados
print(df.columns)

Index(['user_id', 'track', 'artist', 'genre', 'city', 'time', 'day'], dtype='object')


[Volver a Contenidos](#back)

### Valores ausentes <a id='missing_values'></a>
 Primero, encuentra el número de valores ausentes en la tabla. Debes utilizar dos métodos en una secuencia para obtener el número de valores ausentes.

In [10]:
# Calcular el número de valores ausentes
print(df.isna().sum())

user_id       0
track      1343
artist     7567
genre      1198
city          0
time          0
day           0
dtype: int64


No todos los valores ausentes afectan a la investigación. Por ejemplo, los valores ausentes en `track` y `artist` no son cruciales. Simplemente puedes reemplazarlos con valores predeterminados como el string `'unknown'` (desconocido).

Pero los valores ausentes en `'genre'` pueden afectar la comparación entre las preferencias musicales de Springfield y Shelbyville. En la vida real, sería útil saber las razones por las cuales hay datos ausentes e intentar recuperarlos. Pero no tenemos esa oportunidad en este proyecto. Así que tendrás que:
* rellenar estos valores ausentes con un valor predeterminado;
* evaluar cuánto podrían afectar los valores ausentes a tus cómputos;

Reemplazar los valores ausentes en las columnas `'track'`, `'artist'` y `'genre'` con el string `'unknown'`. Como mostramos anteriormente en las lecciones, la mejor forma de hacerlo es crear una lista que almacene los nombres de las columnas donde se necesita el reemplazo. Luego, utiliza esta lista e itera sobre las columnas donde se necesita el reemplazo haciendo el propio reemplazo.

In [11]:
# Bucle en los encabezados reemplazando los valores ausentes con 'unknown'

columns_to_replace=['track', 'artist' , 'genre']
for col in columns_to_replace:
    df[col].fillna('unknown', inplace = True)
    
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65079 entries, 0 to 65078
Data columns (total 7 columns):
 #   Column   Non-Null Count  Dtype 
---  ------   --------------  ----- 
 0   user_id  65079 non-null  object
 1   track    65079 non-null  object
 2   artist   65079 non-null  object
 3   genre    65079 non-null  object
 4   city     65079 non-null  object
 5   time     65079 non-null  object
 6   day      65079 non-null  object
dtypes: object(7)
memory usage: 3.5+ MB
None


Ahora comprueba el resultado para asegurarte de que después del reemplazo no haya valores ausentes en el conjunto de datos. Para hacer esto, cuenta los valores ausentes nuevamente.

In [12]:
# Contar valores ausentes
print(df.isna().sum())

user_id    0
track      0
artist     0
genre      0
city       0
time       0
day        0
dtype: int64


[Volver a Contenidos](#back)

### Duplicados <a id='duplicates'></a>
Encuentra el número de duplicados explícitos en la tabla. Una vez más, debes aplicar dos métodos en una secuencia para obtener la cantidad de duplicados explícitos.

In [13]:
# Contar duplicados explícitos
print(df.duplicated().sum())

3826


Ahora, elimina todos los duplicados. Para ello, llama al método que hace exactamente esto.

In [14]:
# Eliminar duplicados explícitos
print(df.drop_duplicates(inplace=True))

None


Comprobemos ahora si eliminamos con éxito todos los duplicados. Cuenta los duplicados explícitos una vez más para asegurarte de haberlos eliminado todos:

In [15]:
# Comprobar de nuevo si hay duplicados

print(df.duplicated().sum())

0


Ahora queremos deshacernos de los duplicados implícitos en la columna `genre`. Por ejemplo, el nombre de un género se puede escribir de varias formas. Dichos errores también pueden afectar al resultado.

Para hacerlo, primero mostremos una lista de nombres de género únicos, ordenados en orden alfabético. Para ello:
* Extrae la columna `genre` del DataFrame.
* Llama al método que devolverá todos los valores únicos en la columna extraída.


In [16]:
# Inspeccionar los nombres de géneros únicos

generos_unicos = sorted(df['genre'].unique())
print(generos_unicos)
print()
num_generos_unicos = len(generos_unicos)
print(f"Número de géneros únicos antes de la corrección: {num_generos_unicos}")


['acid', 'acoustic', 'action', 'adult', 'africa', 'afrikaans', 'alternative', 'ambient', 'americana', 'animated', 'anime', 'arabesk', 'arabic', 'arena', 'argentinetango', 'art', 'audiobook', 'avantgarde', 'axé', 'baile', 'balkan', 'beats', 'bigroom', 'black', 'bluegrass', 'blues', 'bollywood', 'bossa', 'brazilian', 'breakbeat', 'breaks', 'broadway', 'cantautori', 'cantopop', 'canzone', 'caribbean', 'caucasian', 'celtic', 'chamber', 'children', 'chill', 'chinese', 'choral', 'christian', 'christmas', 'classical', 'classicmetal', 'club', 'colombian', 'comedy', 'conjazz', 'contemporary', 'country', 'cuban', 'dance', 'dancehall', 'dancepop', 'dark', 'death', 'deep', 'deutschrock', 'deutschspr', 'dirty', 'disco', 'dnb', 'documentary', 'downbeat', 'downtempo', 'drum', 'dub', 'dubstep', 'eastern', 'easy', 'electronic', 'electropop', 'emo', 'entehno', 'epicmetal', 'estrada', 'ethnic', 'eurofolk', 'european', 'experimental', 'extrememetal', 'fado', 'film', 'fitness', 'flamenco', 'folk', 'folklor

Busca en la lista para encontrar duplicados implícitos del género `hiphop`. Estos pueden ser nombres escritos incorrectamente o nombres alternativos para el mismo género.

Verás los siguientes duplicados implícitos:
* `hip`
* `hop`
* `hip-hop`

Para deshacerte de ellos, crea una función llamada `replace_wrong_genres()` con dos parámetros:
* `wrong_genres=`: esta es una lista que contiene todos los valores que necesitas reemplazar.
* `correct_genre=`: este es un string que vas a utilizar como reemplazo.

Como resultado, la función debería corregir los nombres en la columna `'genre'` de la tabla `df`, es decir, remplazar cada valor de la lista `wrong_genres` por el valor en `correct_genre`.

Dentro del cuerpo de la función, utiliza un bucle `'for'` para iterar sobre la lista de géneros incorrectos, extrae la columna `'genre'` y aplica el método `replace` para hacer correcciones.

In [17]:
# Función para reemplazar duplicados implícitos

wrong_genres = ['hip','hop','hip-hop']
correct_genre = 'hiphop'



def replace_wrong_genres(df, column_name, wrong_genres, correct_genre):
    for wrong_genre in wrong_genres:
        df[column_name] = df[column_name].replace(wrong_genre, correct_genre)
    return df
  

Ahora, llama a `replace_wrong_genres()` y pásale tales argumentos para que retire los duplicados implícitos (`hip`, `hop` y `hip-hop`) y los reemplace por `hiphop`:

In [18]:
# Eliminar duplicados implícitos
df = replace_wrong_genres(df, 'genre', wrong_genres, correct_genre)



Asegúrate de que los nombres duplicados han sido eliminados. Muestra la lista de valores únicos de la columna `'genre'` una vez más:

In [19]:
# Comprobación de duplicados implícitos
generos_unicos = sorted(df['genre'].unique())
print(generos_unicos)
print()
num_generos_unicos = len(generos_unicos)
print(f"Número de géneros únicos después de la corrección: {num_generos_unicos}")



['acid', 'acoustic', 'action', 'adult', 'africa', 'afrikaans', 'alternative', 'ambient', 'americana', 'animated', 'anime', 'arabesk', 'arabic', 'arena', 'argentinetango', 'art', 'audiobook', 'avantgarde', 'axé', 'baile', 'balkan', 'beats', 'bigroom', 'black', 'bluegrass', 'blues', 'bollywood', 'bossa', 'brazilian', 'breakbeat', 'breaks', 'broadway', 'cantautori', 'cantopop', 'canzone', 'caribbean', 'caucasian', 'celtic', 'chamber', 'children', 'chill', 'chinese', 'choral', 'christian', 'christmas', 'classical', 'classicmetal', 'club', 'colombian', 'comedy', 'conjazz', 'contemporary', 'country', 'cuban', 'dance', 'dancehall', 'dancepop', 'dark', 'death', 'deep', 'deutschrock', 'deutschspr', 'dirty', 'disco', 'dnb', 'documentary', 'downbeat', 'downtempo', 'drum', 'dub', 'dubstep', 'eastern', 'easy', 'electronic', 'electropop', 'emo', 'entehno', 'epicmetal', 'estrada', 'ethnic', 'eurofolk', 'european', 'experimental', 'extrememetal', 'fado', 'film', 'fitness', 'flamenco', 'folk', 'folklor

[Volver a Contenidos](#back)

### Tus observaciones <a id='data_preprocessing_conclusions'></a>

`Describe brevemente lo que has notado al analizar duplicados, cómo abordaste sus eliminaciones y qué resultados obtuviste.`

Al analizar los resultados del Data Frame hubo 3826 datos duplicados explícitos, se uso el método drop_duplicates para eliminarlos,para comprobar si no quedó ninguno usamos el método duplicated junto con sum y el resultado debe ser 0 ya que cuenta el número de valores duplicados. Para el caso de valores duplicados implícitos comprobamos los valores únicos con el método unique en la columna de género de las canciones, el género hiphop presentaba tres formas de escritura hip, hop, hip-hop se corrigió usando una función personalizada para que el data frame solo tenga una forma correcta de llamar al género hiphop, de esa formma los datos pueden ser analizados de forma clara, reduciendo los problemas existentes. 

[Volver a Contenidos](#back)

## Etapa 3. Prueba de hipótesis <a id='hypothesis'></a>

### 3.1 Hipótesis 1: actividad de los usuarios y las usuarias en las dos ciudades <a id='activity'></a>

La hipótesis afirma que existen diferencias en la forma en que los usuarios y las usuarias de Springfield y Shelbyville consumen música. Para comprobar esto, usa los datos de tres días de la semana: lunes, miércoles y viernes.

* Agrupa a los usuarios y las usuarias por ciudad.
* Compara el número de canciones que cada grupo reprodujo el lunes, el miércoles y el viernes.


Realiza cada cálculo por separado.

El primer paso es evaluar la actividad del usuario en cada ciudad. Recuerda las etapas dividir-aplicar-combinar de las que hablamos anteriormente en la lección. Tu objetivo ahora es agrupar los datos por ciudad, aplicar el método apropiado para contar durante la etapa de aplicación y luego encontrar la cantidad de canciones reproducidas en cada grupo especificando la columna para obtener el recuento.

A continuación se muestra un ejemplo de cómo debería verse el resultado final:
`df.groupby(by='....')['column'].method()`Realiza cada cálculo por separado.

Para evaluar la actividad de los usuarios y las usuarias en cada ciudad, agrupa los datos por ciudad y encuentra la cantidad de canciones reproducidas en cada grupo.



In [20]:
# Contar las canciones reproducidas en cada ciudad
filtered_df = df[df['city'].isin(['Springfield', 'Shelbyville'])]
canciones_por_ciudad = filtered_df.groupby(by='city')['track'].count()
print(canciones_por_ciudad)

city
Shelbyville    18512
Springfield    42741
Name: track, dtype: int64


`La ciudad de Springfield tiene más canciones reproducidas que Shelbyville con una diferencia de 24228 canciones reproducidas.`

Ahora agrupemos los datos por día de la semana y encontremos el número de canciones reproducidas el lunes, miércoles y viernes. Utiliza el mismo método que antes, pero ahora necesitamos una agrupación diferente.


In [21]:
# Calcular las canciones reproducidas en cada uno de los tres días
print(df['day'].unique())


def contar_canciones_por_ciudad_y_dia(df, day):
    # Filtrar el DataFrame por el día especificado
    return df[df['day'] == day].groupby('city')['track'].count()

# Calcular las canciones reproducidas en los días lunes, miércoles y viernes
canciones_por_ciudad_lunes = contar_canciones_por_ciudad_y_dia(df, 'Monday')
canciones_por_ciudad_miercoles = contar_canciones_por_ciudad_y_dia(df, 'Wednesday')
canciones_por_ciudad_viernes = contar_canciones_por_ciudad_y_dia(df, 'Friday')

# Mostrar los resultados
print("Canciones por ciudad el lunes:")
print(canciones_por_ciudad_lunes)
print("\nCanciones por ciudad el miércoles:")
print(canciones_por_ciudad_miercoles)
print("\nCanciones por ciudad el viernes:")
print(canciones_por_ciudad_viernes)

['Wednesday' 'Friday' 'Monday']
Canciones por ciudad el lunes:
city
Shelbyville     5614
Springfield    15740
Name: track, dtype: int64

Canciones por ciudad el miércoles:
city
Shelbyville     7003
Springfield    11056
Name: track, dtype: int64

Canciones por ciudad el viernes:
city
Shelbyville     5895
Springfield    15945
Name: track, dtype: int64


`Al revisar los resultados, la ciudad de Springfield es la que ha reproducido el mayor número de canciones los días lunes, miércoles y viernes, en comparación con la ciudad de Shelbyville . El día que se reproducen más canciones en Springfield es el viernes con 15945 canciones reproducidas, pero el día que más se reproducen canciones en Shelbyville es el miércoles con 7003 canciones reproducidas y su día con menos reproducciones es el lunes, en cambio para Springfield su día con menos reproducciones es el miércoles.`

Ya sabes cómo contar entradas agrupándolas por ciudad o día. Ahora necesitas escribir una función que pueda contar entradas según ambos criterios simultáneamente.

Crea la función `number_tracks()` para calcular el número de canciones reproducidas en un determinado día **y** ciudad. La función debe aceptar dos parámetros:

- `day`: un día de la semana para filtrar. Por ejemplo, `'Monday'` (lunes).
- `city`: una ciudad para filtrar. Por ejemplo, `'Springfield'`.

Dentro de la función, aplicarás un filtrado consecutivo con indexación lógica.

Primero filtra los datos por día y luego filtra la tabla resultante por ciudad.

Después de filtrar los datos por dos criterios, cuenta el número de valores de la columna 'user_id' en la tabla resultante. Este recuento representa el número de entradas que estás buscando. Guarda el resultado en una nueva variable y devuélvelo desde la función.

In [22]:
# Declara la función number_tracks() con dos parámetros: day= y city=.    
def number_tracks(df, day, city):

    # Almacena las filas del DataFrame donde el valor en la columna 'day' es igual al parámetro day=
    filtered_by_day = df[df['day'] == day]
    
    # Filtra las filas donde el valor en la columna 'city' es igual al parámetro city=
    
    filtered_by_city = filtered_by_day[filtered_by_day['city'] == city]
    
    # Extrae la columna 'user_id' de la tabla filtrada y aplica el método count()
    count_tracks = filtered_by_city['user_id'].count()
    
    # Devolve el número de valores de la columna 'user_id'
    return count_tracks


Llama a `number_tracks()` seis veces, cambiando los valores de los parámetros para que recuperes los datos de ambas ciudades para cada uno de los tres días.

In [23]:
# El número de canciones reproducidas en Springfield el lunes
canciones_lunes_springfield = number_tracks(df, 'Monday', 'Springfield')
print(f"Número de canciones reproducidas en Springfield el lunes: {canciones_lunes_springfield}")


Número de canciones reproducidas en Springfield el lunes: 15740


In [24]:
# El número de canciones reproducidas en Shelbyville el lunes
canciones_lunes_shelbyville = number_tracks(df, 'Monday', 'Shelbyville')
print(f"Número de canciones reproducidas en Shelbyville el lunes: {canciones_lunes_shelbyville}")


Número de canciones reproducidas en Shelbyville el lunes: 5614


In [25]:
# El número de canciones reproducidas en Springfield el miércoles
canciones_miercoles_springfield = number_tracks(df, 'Wednesday', 'Springfield')
print(f"Número de canciones reproducidas en Springfield el miércoles: {canciones_miercoles_springfield}")


Número de canciones reproducidas en Springfield el miércoles: 11056


In [26]:
# El número de canciones reproducidas en Shelbyville el miércoles
canciones_miercoles_shelbyville = number_tracks(df, 'Wednesday', 'Shelbyville')
print(f"Número de canciones reproducidas en Shelbyville el miércoles: {canciones_miercoles_shelbyville}")

Número de canciones reproducidas en Shelbyville el miércoles: 7003


In [27]:
# El número de canciones reproducidas en Springfield el viernes
canciones_viernes_springfield = number_tracks(df, 'Friday', 'Springfield')
print(f"Número de canciones reproducidas en Springfield el viernes: {canciones_viernes_springfield}")


Número de canciones reproducidas en Springfield el viernes: 15945


In [28]:
# El número de canciones reproducidas en Shelbyville el viernes
canciones_viernes_shelbyville = number_tracks(df, 'Friday', 'Shelbyville')
print(f"Número de canciones reproducidas en Shelbyville el viernes: {canciones_viernes_shelbyville}")

Número de canciones reproducidas en Shelbyville el viernes: 5895


**Conclusiones**

`Comenta si la hipótesis es correcta o se debe rechazar . Explica tu razonamiento.`
La hipótesis afirma que existen diferencias en la forma en que los usuarios y las usuarias de Springfield y Shelbyville consumen música. Según los datos obtenidos la hipótesis es correcta porque sí existen diferencias significativas entre el número de canciones reproducidas entre las ciudades de Shelbyville y Springfield. Al analizar a Springfield esta es la ciudad que tiene más reproducciones musicales por usuario y por número de canciones comparación con Shelbyville.
El día que más reproduce canciones Springfield es el viernes (15945 canciones), en comparación con Shelbyville que es el día lunes(7003 canciones). 

[Volver a Contenidos](#back)

# Conclusiones <a id='end'></a>

`Resume aquí tus conclusiones sobre la hipótesis.`
1. Según los datos obtenidos la hipótesis es correcta porque sí existen diferencias significativas entre el número de canciones reproducidas entre las ciudades de Shelbyville y Springfield.
2. El día lunes Springfield reproduce 10126 canciones más que Shelbyville, el día miércoles 4053 canciones más y el día viernes 1050 canciones más .
3. El día que más reproduce canciones Springfield es el viernes (15945 canciones), en comparación con Shelbyville que es el día miércoles(7003 canciones). 
4. El día que menos reproduce canciones Springfield es el miércoles (11056 canciones), en comparación con Shelbyville que es el día lunes(5614 canciones). 

### Nota
En proyectos de investigación reales, la prueba de hipótesis estadística es más precisa y cuantitativa. También ten en cuenta que no siempre se pueden sacar conclusiones sobre una ciudad entera a partir de datos de una sola fuente.

Aprenderás más sobre la prueba de hipótesis en el sprint de análisis estadístico de datos.

[Volver a Contenidos](#back)